### PR TITLE
Provide generic interfaces for token based authorization

### PR DESCRIPTION
**Pull request by raakella1**
Replace the specific implementation of token based authorization with generic interfaces.

### CODE REVIEW

File: include/sisl/auth_manager/token_client.hpp [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+    virtual std::string get_token() = 0;
+};
+
+// the key value pairs (m_auth_header_key, get_token()) are sent as metadata in the grpc client context
+
+class GrpcTokenClient : public TokenClient {
+public:
+    explicit GrpcTokenClient(std::string const& auth_header_key) : m_auth_header_key(auth_header_key) {}
+    virtual ~GrpcTokenClient() = default;
```
Comment (szmyd) [inline include/sisl/auth_manager/token_client.hpp:20]: `override`?

File: include/sisl/auth_manager/token_client.hpp [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace sisl {
+
+// Interface to get a token for authorization
+
+class TokenClient {
+public:
+    virtual ~TokenClient() = default;
+
+    virtual std::string get_token() = 0;
```
Comment (szmyd) [inline include/sisl/auth_manager/token_client.hpp:12]: `const`

**Review by szmyd — COMMENTED** [review]
Not clear to me why there are _two_ interfaces here?

File: include/sisl/auth_manager/token_client.hpp (hunk 2 above)
Comment (raakella1) [reply include/sisl/auth_manager/token_client.hpp:12]: This is only ever implemented by Mock for unit tests. The `real` implementation will be in the new library I created, which is ebay specific.
https://github.corp.ebay.com/SDS/auth_manager/tree/inherit_from_sisl/src/include/auth_manager
The idea is if the user wants a token based auth system, they implement these interfaces and pass it along to grpc client and server. Any implementation would be ebay specific and cannot live here in sisl

**Review by szmyd — APPROVED** [review]
lgtm

**Pull request merged**